src/main/java/org/zalando/nakadi/service/FeatureToggleServiceZk.java

```diff
@@ -1,57 +1,55 @@
-package org.zalando.nakadi.util;
+package org.zalando.nakadi.service;
 
-import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.CacheLoader;
-import com.google.common.cache.LoadingCache;
 import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
 import org.apache.zookeeper.KeeperException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
 
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
+import javax.annotation.PreDestroy;
+import java.io.IOException;
 
 public class FeatureToggleServiceZk implements FeatureToggleService {
 
     private static final Logger LOG = LoggerFactory.getLogger(FeatureToggleService.class);
-    private static final String PREFIX = "/nakadi/feature_toggle/";
+    private static final String PREFIX = "/nakadi/feature_toggle";
 
     private final ZooKeeperHolder zkHolder;
-    private final LoadingCache<Feature, Boolean> cache;
+    private PathChildrenCache featuresCache;
 
     public FeatureToggleServiceZk(final ZooKeeperHolder zkHolder) {
         this.zkHolder = zkHolder;
+        try {
+            this.featuresCache = new PathChildrenCache(zkHolder.get(), PREFIX, false);
+            this.featuresCache.start();
+        } catch (final Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
 
-        cache = CacheBuilder.newBuilder()
-                .maximumSize(100)
-                .expireAfterWrite(30, TimeUnit.SECONDS)
-                .build(new CacheLoader<Feature, Boolean>() {
-                    @Override
-                    public Boolean load(final Feature key) throws Exception {
-                        return isFeatureEnabledInZk(key);
-                    }
-                });
-
+    @PreDestroy
+    public void cleanUp() {
+        try {
+            this.featuresCache.close();
+        } catch (IOException e) {
+            LOG.error("Could not close features cache", e);
+        }
     }
 
     public boolean isFeatureEnabled(final Feature feature) {
         try {
-            return cache.get(feature);
-        } catch (final ExecutionException e) {
+            return featuresCache.getCurrentData(PREFIX + "/" + feature.getId()) != null;
```

Comment:
I thought about create the path beforehand, since there is limit amount of paths, but from other side it could be optimised by jvm

Reply:
Doesn't take much to create the paths beforehand, so I did that